DuckDuckGo/TabManager.swift

```diff
@@ -123,7 +123,21 @@ class TabManager {
         model.select(tabAt: model.count - 1)
         save()
     }
-    
+
+    func firstHomeTab() -> Tab? {
+        return model.tabs.first(where: { $0.link == nil })
+    }
+
+    func first(withUrl url: URL) -> Tab? {
+        return model.tabs.first(where: { $0.link?.url == url })
```

Comment:
An edge case here is if you've opened a site that is then upgraded to HTTPS, and later open the HTTP link again. Is it worth reusing the HTTPS tab?

Reply:
Good point - I'll fix that before merging.

Reply:
I've just pushed that.  Any chance you could give that a wee test, please? 

Also, I made a small fix to improve favicon loading for favorites, so if you could keep an eye on that while you do that, that would be great, thanks!

Reply:
This is looking good to me now, nice work @brindy!

I found one new edge case where I tested with a local restaurant's website which I opened via Apple Maps, but the HTTPS redirected version adds a trailing forward slash, so it still fails.

```
(lldb) po linkUrl
▿ https://www.thevillageidiot.ca/
  - _url : https://www.thevillageidiot.ca/

(lldb) po url
▿ http://www.thevillageidiot.ca
  - _url : http://www.thevillageidiot.ca
```

I think this is getting into edge case of edge case territory, so I'm honestly fine with the current behaviour.

Reply:
Hah, OK thanks.  It's better than it was for sure!